src/init.jl

```diff
@@ -95,17 +99,22 @@ function __init__()
         #     return v, g
         # end
 
-        function Hamiltonian(metric, ℓπ)
+        function Hamiltonian(metric::AbstractMetric, ℓπ, ::Union{ForwardDiffAD, Type{ForwardDiffAD}})
             ∂ℓπ∂θ(θ::AbstractVecOrMat) = ∂ℓπ∂θ_forwarddiff(ℓπ, θ)
             return Hamiltonian(metric, ℓπ, ∂ℓπ∂θ)
-        end
+        end        
+
+        export ForwardDiffAD

```

Comment:
ForwardDiffAD exists in Turing:
https://github.com/TuringLang/Turing.jl/blob/aad1aee6b66095edaadcc9eb05129cd882609a25/src/core/ad.jl#L28
So either it should not be exported or it should be imported in Turing, I guess.

Reply:
AHMC is not dependent on Turing. This PR is not internal. It's utilites for users who wants to interact with AHMC directly instead of using it through Turing.

Reply:
There is no change for Turing because the AD control is explicty there when constructing `Hamiltonian`.

Reply:
Oh sorry I just got your point of not exporting. Let me think of it.

Reply:
Sure, I see why you want to export it and that Turing does not use the convenience constructor. I was just wondering if the singletons of the same name that exist in Turing might lead to problems.